plutus-use-cases/src/Language/PlutusTx/Coordination/Contracts/Escrow.hs

```diff
@@ -61,30 +62,30 @@ type EscrowSchema =
         .\/ Endpoint "refund-escrow" ()
 
 -- $escrow
--- The escrow contract implements the exchange of value between multiple 
--- parties. It is defined by a list of targets (public keys and script 
--- addresses, each associated with a value). It works similar to the 
+-- The escrow contract implements the exchange of value between multiple
+-- parties. It is defined by a list of targets (public keys and script
+-- addresses, each associated with a value). It works similar to the
 -- crowdfunding contract in that the contributions can be made independently,
 -- and the funds can be unlocked only by a transaction that pays the correct
--- amount to each target. A refund is possible if the outputs locked by the 
+-- amount to each target. A refund is possible if the outputs locked by the
 -- contract have not been spent by the deadline. (Compared to the crowdfunding
 -- contract, the refund policy is simpler because here because there is no
 -- "collection period" during which the outputs may be spent after the deadline
--- has passed. This is because we're assuming that the participants in the 
--- escrow contract will make their deposits as quickly as possible after 
+-- has passed. This is because we're assuming that the participants in the
+-- escrow contract will make their deposits as quickly as possible after
 -- agreeing on a deal)
--- 
--- The contract supports two modes of operation, manual and automatic. In 
+--
+-- The contract supports two modes of operation, manual and automatic. In
 -- manual mode, all actions are driven by endpoints that exposed via 'payEp'
--- 'redeemEp' and 'refundEp'. In automatic mode, the 'pay', 'redeem' and 
--- 'refund'actions start immediately. This mode is useful when the escrow is 
+-- 'redeemEp' and 'refundEp'. In automatic mode, the 'pay', 'redeem' and
+-- 'refund'actions start immediately. This mode is useful when the escrow is
 -- called from within another contract, for example during setup (collection of
 -- the initial deposits).
 
 -- | Defines where the money should go.
 data EscrowTarget =
     PubKeyTarget PubKey Value
-    | ScriptTarget ValidatorHash DataScript Value
+    | ScriptTarget ValidatorHash PubKey Value
```

Comment:
Hm, but then we'd need an `HList` of `EscrowTarget`, if we want to pay into contracts with different types of data script.

The current version of `EscrowTarget` is nice because it directly corresponds to transaction outputs.

Reply:
Ok, how about I use the escrow in one of the financial contracts as I had intended, and then we get back to this PR?

Reply:
Good plan :)